bin/cask

```diff
@@ -324,10 +330,21 @@ def main():
             exit_error(
                 'Python 2.6 required, yours is {0.major}.{0.minor}'.format(
                     sys.version_info))
+
+
         if len(sys.argv) > 1 and sys.argv[1] == 'exec':
+            ## never returns!
             exec_command(sys.argv[2:])
-        else:
-            exec_cask(sys.argv[1:])
+
+        ## special handling for --emacs/bin as this has to happen before
+        ## everything else, and we don't want the parsing in two places.
+        if len(sys.argv) > 1:
+            if sys.argv[1] == '--emacs':
+                global EMACS
```

Comment:
Okay -- I'll need to refactor `exit_error` later on I think, which depends on sys.argv.

Is there a reason for `get_cask_emacs` -- it can be run several times (via `get_cask_path`) and each times requires a full Emacs invocation. Would it not make more sense to set this once an init time?

Would you still be unhappy with the use of a global EMACS variable -- defined once in main, rather similar to the CASK* variables (which are also globals).